src/maison/utils.py

```diff
@@ -109,3 +109,43 @@ def _collect_configs(
             sources.append(IniSource(**source_kwargs))
 
     return sources
+
+
+def deep_merge(destination: Dict[Any, Any], source: Dict[Any, Any]) -> Dict[Any, Any]:
+    """Recursively updates the destination dictionary.
+
+    Args:
+        destination: A dictionary to be merged into. This will be updated in place.
+        source: The dictionary supplying data
+
+    Returns:
+        The updated destination dictionary.
+
+    Raises:
+        RuntimeError: A dict cannot be merged on top of a non-dict.
+          For example, the following would fail:
+          `deep_merge({"foo": "bar"}, {"foo": {"baz": "qux"}})`
+
+    Usage example:
+    >>> a = { 'first' : { 'all_rows' : { 'pass' : 'dog', 'number' : '1' } } }
+    >>> b = { 'first' : { 'all_rows' : { 'fail' : 'cat', 'number' : '5' } } }
+    >>> deep_merge(a, b) == {
+    ...     "first": {"all_rows": {"pass": "dog", "fail": "cat", "number": "5"}}
+    ... }
+    True
+
+    Based on https://stackoverflow.com/a/20666342
```

Comment:
i think this block should be above the `Args:` section

Reply:
Done in https://github.com/dbatten5/maison/pull/89/commits/46d7f04f1268fbca5aee61efe7571fddb3d171b4.